controller/src/beerocks/master/son_actions.cpp

```diff
@@ -154,14 +156,27 @@ void son_actions::unblock_sta(db &database, ieee1905_1::CmduMessageTx &cmdu_tx,
             continue;
         }
         Socket *sd = database.get_node_socket(hostap);
-        auto request =
-            message_com::create_vs_message<beerocks_message::cACTION_CONTROL_CLIENT_ALLOW_REQUEST>(
-                cmdu_tx);
-        if (request == nullptr) {
-            LOG(ERROR) << "Failed building ACTION_CONTROL_CLIENT_ALLOW_REQUEST message!";
+        if (!cmdu_tx.create(0,
+                            ieee1905_1::eMessageType::CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE)) {
+            LOG(ERROR)
+                << "cmdu creation of type CLIENT_ASSOCIATION_CONTROL_REQUEST_MESSAGE, has failed";
             break;
         }
-        request->mac() = network_utils::mac_from_string(sta_mac);
+        auto association_control_request_tlv =
+            cmdu_tx.addClass<wfa_map::tlvClientAssociationControlRequest>();
+        if (!association_control_request_tlv) {
+            LOG(ERROR) << "addClass wfa_map::tlvClientAssociationControlRequest failed";
+            break;
+        }
+        association_control_request_tlv->bssid_to_block_client() =
+            network_utils::mac_from_string(current_ap_mac);
+        association_control_request_tlv->association_control() =
+            wfa_map::tlvClientAssociationControlRequest::UNBLOCK;
+        //TODO: Get real validity_period_sec
```

Comment:
I think this is only valid for blocking, not for unblocking, so it could be set to 0. But I don't find the relevant part of the MultiAP spec.

Reply:
I think you are right :
`The Validity Period field in a Client Association Control Request message with Association Control field set to 0x01 (Client Unblocking) is ignored`  
So I'll change its value to zero.